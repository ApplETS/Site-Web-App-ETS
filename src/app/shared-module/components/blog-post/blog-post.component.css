.container {
    display: flex;
    flex-direction: column;
    border: 1px solid white;
    border-radius: 5px;
    background-color: #0003;
    box-shadow: 0 10px 16px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19) !important;
    color: #ffffff;
    margin: 40px auto 40px auto;
}

.content-container {
    display: flex;
    justify-content: center;
}

.image {
    border-radius: 5px;
}

.news-text-container {
    display: flex;
    align-items: center;
    text-align: justify;
}

.news-link-container {
    display: flex;
    justify-content: center;
    margin-bottom: 20px;
}

.news-button-container {
    display: flex;
    align-items: center;
    padding: 10px;
    border: 1px solid rgba(255, 255, 255, 0.534);
    border-radius: 5px;
    cursor: pointer;
    text-decoration-line: none;
    transition: background-color 300ms;
    color: white;
}

.news-button-container:hover {
    background-color: white;
    color: #2C2E2F;
}

.title {
    text-align: center;
}

.main-title {
    font-size: 2.2rem;
}

.sub-title {
    font-size: 1.1rem;
}

.sub-sub-title {
    font-size: 0.9rem;
}

@media (min-width: 576px) {
    .container {
        max-width: 540px;
    }
}

@media (min-width: 768px) {
    .container {
        max-width: 720px;
    }
}

@media (min-width: 992px) {
    .container {
        max-width: 960px;
    }

    .description-content {
        height: 220vh;
    }

    .content-container {
        margin: 40px;
    }

    .news-text-container {
        margin-bottom: 20px;
    }

    .link-button {
        font-size: 1rem;
        margin-top: 0.7rem;
    }

    .news-button-container {
        font-size: 0.8rem;
    }
}

@media (min-width: 1200px) {
    .container {
        max-width: 1140px;
    }
}
