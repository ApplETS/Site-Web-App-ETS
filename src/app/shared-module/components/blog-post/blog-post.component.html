<div class="container">
    <span class="title main-title">{{post.title}}</span>
    <span class="title sub-title">{{post.subtitle}}</span>
    <span class="title sub-sub-title">{{post.date | date:'fullDate'}}</span>

    <div class="content-container">
        <div *ngIf="post.text">{{post.text}}</div>
        <iframe
                *ngIf="post.iFrameUrl"
                [src]="sanitizer.bypassSecurityTrustResourceUrl(post.iFrameUrl)"
                width="500"
                height="481"
                style="border:none;overflow:hidden"
                scrolling="no"
                frameborder="0"
                allowTransparency="true"
                allow="encrypted-media">
        </iframe>
    </div>

    <div class="news-link-container" *ngIf="post.bottomButtons">
        <app-go-to-button *ngFor="let button of post.bottomButtons"
                          link="{{button.link}}"
                          text="{{button.text}}"
                          icon="{{button.icon}}"
                          iconAlt="{{button.iconAlt}}"
        ></app-go-to-button>
    </div>
</div>
