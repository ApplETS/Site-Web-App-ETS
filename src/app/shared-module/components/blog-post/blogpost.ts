export class BlogPost {
  title: string;
  subtitle: string;
  date: Date;

  // optional
  location?: string;
  text?: string;
  iFrameUrl?: string;

  bottomButtons?: any[] = [];

  constructor(title: string,
              subtitle: string,
              date: Date,
              location?: string,
              text?: string,
              iFrameUrl?: string,
              bottomButtons?: any[]) {
    this.title = title;
    this.subtitle = subtitle;
    this.date = date;
    this.location = location;
    this.text = text;
    this.iFrameUrl = iFrameUrl;
    this.bottomButtons = bottomButtons;
  }
}
