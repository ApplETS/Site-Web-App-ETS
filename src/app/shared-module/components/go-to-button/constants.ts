export const LOGO_FOLDER = 'assets/images/logos';
export const FACEBOOK_ICON = LOGO_FOLDER + '/facebook-logo.png';
export const TWITTER_ICON = LOGO_FOLDER + '/twitter-logo.png';
export const GITHUB_OCTOCAT_ICON = LOGO_FOLDER + '/github-white-octocat-logo.png';
export const GITHUB_ICON = LOGO_FOLDER + '/github-logo.png';

export const FACEBOOK_LABEL = 'Facebook logo';
export const TWITTER_LABEL = 'Twitter logo';
export const GITHUB_LABEL = 'Github logo';
