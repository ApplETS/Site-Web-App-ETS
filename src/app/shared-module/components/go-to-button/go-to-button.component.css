.container{
    display: flex;
    align-items: center;
    padding: 10px;
    border: 1px solid rgba(255, 255, 255, 0.534);
    border-radius: 5px;
    cursor:pointer;
    text-decoration-line: none;
    transition:background-color 300ms;
    color:white;
    width: fit-content;
}

.container:hover{
    background-color:white;
    color:#2C2E2F;
}

.logo-image{
    margin-right:10px;
    height:2rem;
    border-radius:3px;
}

@media (max-width: 992px) {
    .logo-image{
        height:1.5rem;
    }
}
