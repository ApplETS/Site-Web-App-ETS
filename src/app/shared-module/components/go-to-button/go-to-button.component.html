<div>
    <a class="container" href='{{link}}'>
        <img class="logo-image" src='{{icon}}' alt='{{iconAlt}}'>
        <span style="text-decoration-line:none">{{text}}</span>
    </a>
    <div class="spacer-horizontal"></div>
</div>
