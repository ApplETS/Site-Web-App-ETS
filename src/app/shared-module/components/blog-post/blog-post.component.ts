import { Component, Input, OnInit } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';

import { BlogPost } from './blogpost';

@Component({
  selector: 'app-blog-post',
  templateUrl: './blog-post.component.html',
  styleUrls: ['./blog-post.component.css']
})
export class BlogPostComponent implements OnInit {
  @Input() post: BlogPost;

  constructor(private sanitizer: DomSanitizer) { }

  ngOnInit() {}
}
