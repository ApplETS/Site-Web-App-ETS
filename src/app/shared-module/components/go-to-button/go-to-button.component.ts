import { Component, Input, OnInit } from '@angular/core';
import * as constants from './constants';

@Component({
  selector: 'app-go-to-button',
  templateUrl: './go-to-button.component.html',
  styleUrls: ['./go-to-button.component.css']
})
export class GoToButtonComponent implements OnInit {

  @Input() link: string;

  @Input() text: string;

  /**
   * The icon is automatically choose if the link is from facebook, twitter or github
   */
  @Input() icon?: string;
  @Input() iconAlt?: string;

  ngOnInit() {
    if (!this.icon && this.link) {
      if (this.link.includes('facebook')) {
        this.icon = constants.FACEBOOK_ICON;
        this.iconAlt = constants.FACEBOOK_LABEL;
      } else if (this.link.includes('twitter')) {
        this.icon = constants.TWITTER_ICON;
        this.iconAlt = constants.TWITTER_LABEL;
      } else if (this.link.includes('github')) {
        this.iconAlt = constants.GITHUB_LABEL;
        if (this.text !== '') {
          this.icon = constants.GITHUB_OCTOCAT_ICON;
        } else {
          this.icon = constants.GITHUB_ICON;
        }
      }
    }
  }

}
