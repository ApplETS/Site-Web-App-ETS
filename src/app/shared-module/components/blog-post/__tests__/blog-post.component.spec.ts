import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ChangeDetectionStrategy } from '@angular/core';
import { ConfigureFn, configureTests } from '../../../../../config.helpers';

import { BlogPostComponent } from '../blog-post.component';
import { BlogPost } from '../blogpost';
import { GoToButtonComponent } from '../../go-to-button/go-to-button.component';

describe('BlogPostComponent', () => {
  let component: BlogPostComponent;
  let fixture: ComponentFixture<BlogPostComponent>;

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [BlogPostComponent, GoToButtonComponent],

      })
        .overrideComponent(BlogPostComponent, {
          set: { changeDetection: ChangeDetectionStrategy.Default }
        });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(BlogPostComponent);
      component = fixture.componentInstance;

      component.post = new BlogPost('testTitle',
        'testSubtitle',
        new Date('03/23/2019'),
        null,
        'Test'
      );
      fixture.detectChanges();
    });
  }));

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should contains a location', () => {
    component.post = new BlogPost('testTitle',
      'testSubtitle',
      new Date('03/23/2019'),
      'Test location',
      'Test'
    );
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should contains a iFrame', () => {
    component.post = new BlogPost('testTitle',
      'testSubtitle',
      new Date('03/23/2019'),
      null,
      null,
      'https://github.com');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should contains a link button', () => {
    component.post = new BlogPost('testTitle',
      'testSubtitle',
      new Date('03/23/2019'),
      null,
      null,
      null,
      [{
        links: 'https://test.test',
        text: 'Voir test'
      }]);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
