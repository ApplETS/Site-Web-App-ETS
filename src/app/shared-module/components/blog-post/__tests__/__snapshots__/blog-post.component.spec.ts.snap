// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostComponent should contains a iFrame 1`] = `
<app-blog-post
  post={[Function BlogPost]}
  sanitizer={[Function DomSanitizerImpl]}
>
  <div
    class="container"
  >
    <span
      class="title main-title"
    >
      testTitle
    </span>
    <span
      class="title sub-title"
    >
      testSubtitle
    </span>
    <span
      class="title sub-sub-title"
    >
      Saturday, March 23, 2019
    </span>
    <div
      class="content-container"
    >
      
      
      <iframe
        allow="encrypted-media"
        allowtransparency="true"
        frameborder="0"
        height="481"
        scrolling="no"
        src="https://github.com"
        style="border:none;overflow:hidden"
        width="500"
      />
    </div>
    
  </div>
</app-blog-post>
`;

exports[`BlogPostComponent should contains a link button 1`] = `
<app-blog-post
  post={[Function BlogPost]}
  sanitizer={[Function DomSanitizerImpl]}
>
  <div
    class="container"
  >
    <span
      class="title main-title"
    >
      testTitle
    </span>
    <span
      class="title sub-title"
    >
      testSubtitle
    </span>
    <span
      class="title sub-sub-title"
    >
      Saturday, March 23, 2019
    </span>
    <div
      class="content-container"
    >
      
      
    </div>
    
    <div
      class="news-link-container"
    >
      
      <app-go-to-button
        ng-reflect-icon=""
        ng-reflect-icon-alt=""
        ng-reflect-link=""
        ng-reflect-text="Voir test"
      >
        <div>
          <a
            class="container"
            href=""
          >
            <img
              alt=""
              class="logo-image"
              src=""
            />
            <span
              style="text-decoration-line:none"
            >
              Voir test
            </span>
          </a>
          <div
            class="spacer-horizontal"
          />
        </div>
      </app-go-to-button>
    </div>
  </div>
</app-blog-post>
`;

exports[`BlogPostComponent should contains a location 1`] = `
<app-blog-post
  post={[Function BlogPost]}
  sanitizer={[Function DomSanitizerImpl]}
>
  <div
    class="container"
  >
    <span
      class="title main-title"
    >
      testTitle
    </span>
    <span
      class="title sub-title"
    >
      testSubtitle
    </span>
    <span
      class="title sub-sub-title"
    >
      Saturday, March 23, 2019
    </span>
    <div
      class="content-container"
    >
      
      <div>
        Test
      </div>
      
    </div>
    
  </div>
</app-blog-post>
`;

exports[`BlogPostComponent should match snapshot 1`] = `
<app-blog-post
  post={[Function BlogPost]}
  sanitizer={[Function DomSanitizerImpl]}
>
  <div
    class="container"
  >
    <span
      class="title main-title"
    >
      testTitle
    </span>
    <span
      class="title sub-title"
    >
      testSubtitle
    </span>
    <span
      class="title sub-sub-title"
    >
      Saturday, March 23, 2019
    </span>
    <div
      class="content-container"
    >
      
      <div>
        Test
      </div>
      
    </div>
    
  </div>
</app-blog-post>
`;
